### PR TITLE
Move CLI to Cobra and add Viper + Small Refactor

### PR DESCRIPTION
**Pull request by mplachter**
Changes
* [viper](https://github.com/spf13/viper)
  * Allow for expanding out of configuration to not only support CLI args but env variables as well.
* [cobra](https://github.com/spf13/cobra)
  * Moved to cobra to give the CLI a more production feel
* Refactored the project
  * While adding cobra and viper the project was getting a little hard to follow the separate domains, wanted to split this up a little bit to make sure this was segmented out more

### CODE REVIEW

File: metrics/aggregate.go [hunk 1]
```diff
@@ -210,11 +215,11 @@ func (a *aggregate) handleInsert(c *gin.Context) {
 	c.Status(http.StatusAccepted)
 }
 
-type labelPair struct {
+type LabelPair struct {
```
Comment (djeebus) [inline metrics/aggregate.go:218]: why expose this? it's not used outside of this package

File: routers/public.go [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+	"github.com/zapier/prom-aggregation-gateway/metrics"
 )
 
-type apiRouterConfig struct {
-	corsDomain string
-	accounts   gin.Accounts
+type ApiRouterConfig struct {
+	CorsDomain   string
+	Accounts     []string
+	authAccounts gin.Accounts
```
Comment (djeebus) [inline routers/public.go:15]: doesn't look like this needs to be part of the config; we could just parse and pass the results into `gin.BasicAuth` when needed. that also helps prevent the Split Brain problem on lines 50-51 (we check `cfg.Accounts`, then use `cfg.authAccounts`)

File: config/config.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			return err
+		}
+	}
+
+	v.SetEnvPrefix(envPrefix)
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	v.AutomaticEnv()
+	bindFlags(cmd, v)
+
+	return nil
```
Comment (djeebus) [inline config/config.go:49]: :ok_hand:

File: cmd/version.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+var initCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Show version information",
+	Long:  `Prints out the version and commit short sha information`,
+	Run:   versionFunc,
+}
+
+func versionFunc(cmd *cobra.Command, args []string) {
+	log.Printf("\n%s\nVersion: %s\nCommitSHA: %s", config.Name, config.Version, config.CommitSHA)
+}
```
Comment (djeebus) [inline cmd/version.go:23]: Adding two subcommands seems heavy when a `--version` would have worked as well. I suspect we're not going to get a lot of other subcommands here :shrug:

**Review by djeebus — DISMISSED** [review]

File: metrics/aggregate.go (hunk 1 above)
Comment (mplachter) [reply metrics/aggregate.go:218]: I was going to use it externally during the refactor but i decded to refactor the unit tests instead for the aggergator to not use the router. Good catch i'll move this back private :)

File: routers/public.go (hunk 2 above)
Comment (mplachter) [reply routers/public.go:15]: We could. I like doing this incase we need to use the accounts anywhere else in the future

File: cmd/version.go (hunk 4 above)
Comment (mplachter) [reply cmd/version.go:23]: I rather have a sub command than some logic that says if this flag is not false than do this and return. It makes it easier to read and maintain

**Review by djeebus — APPROVED** [review]
